### PR TITLE
android: handle onDestroy

### PR DESCRIPTION
**Pull request by pazos**
We need it to exit the lua state and return from android_main. This avoid a *zombie state** on the following scenarios:

- App is killed by the activity manager  (ie: battery optimizations or OOM)
- We call finish() from Kotlin

**zombie state**: app is running, does nothing, won't easily trigger ANRs and needs to be killed manually by the user

In the real world the only related bug I found was on android10+, when running KO ocassionally and leaving the activity in the background for a week. The app is then destroyed by the system to "save battery".

As android has a way to "upgrade" or "downgrade" app battery optimization based on usage, YMMV.

Requires https://github.com/koreader/koreader-base/pull/1224

---
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/6811)

### CODE REVIEW

**Comment by pazos** [comment]
There's no much to review here, but I will wait for an approval :smile:

File: frontend/device/android/device.lua
Comment (NiLuJe) [inline frontend/device/android/device.lua:139]: Would a UIManager:quit or whatever the Exit menu does make sense here?

Comment (pazos) [reply frontend/device/android/device.lua]: Yeah, makes all the sense. It is also slightly more verbose :)

**Review by poire-z — APPROVED** [review]

**Pull request merged**
